ha/core/event_analyzer/watcher/watcher.py

```diff
@@ -0,0 +1,125 @@
+
+from cortx.utils.message_bus import MessageConsumer
+from cortx.utils.log import Log
+
+from ha.core.event_analyzer.filter.filter import Filter
+from ha.core.event_analyzer.parser.parser import Parser
+from ha.core.event_analyzer.subscriber import Subscriber
+from ha.core.event_analyzer.event_analyzer_exceptions import InvalidSubscriber
+from ha.core.event_analyzer.event_analyzer_exceptions import EventFilterException
+from ha.core.event_analyzer.event_analyzer_exceptions import EventParserException
+from ha.core.event_analyzer.event_analyzer_exceptions import SubscriberException
+
+class Watcher(Thread):
+    """ Watch message bus to check in coming event. """
+
+    def __init__(self, consumer_id: int, message_type: str, consumer_group: str,
+                event_filter: Filter, event_parser: Parser, subscriber: Subscriber):
+        """
+        Initalize Watcher class to monitor message bus event.
+
+        Args:
+            id (int): Consumer ID for message bus.
+            message_type (str): Message type for getting event.
+            group (str): Consumer Group of message bus.
+            event_filter (Filter): Filter unused event.
+            parser (Parser): Parse event to HealthEvent
+            subscriber (Subscriber): Pass event to Subscriber.
+        """
+        super(Watcher, self).__init__(name=f"{message_type}-{str(consumer_id)}", daemon=True)
+        Log.info(f"Initalizing watcher {message_type}-{str(consumer_id)}")
+        self.consumer_id = consumer_id
+        self.message_type = message_type
+        self.consumer_group = consumer_group
+        self.filter = event_filter
+        self.parser = event_parser
+        self.subscriber = subscriber
+        self._validate()
+        self.consumer = self._get_connection()
+
+    def _validate(self) -> None:
+        """
+        Validate watcher and raise exception.
+
+        Raises:
+            EventAnalyzer: event analyser exception.
+        """
+        if not isinstance(self.subscriber, Subscriber):
+            raise InvalidSubscriber(f"Invaid subscriber {self.subscriber}")
+
+    def _get_connection(self) -> MessageConsumer:
+        """
+        Get message consumer connection.
+
+        Returns:
+            MessageConsumer: Return instance of MessageConsumer.
+        """
+        return MessageConsumer(consumer_id=str(self.consumer_id),
+                                consumer_group=self.consumer_group,
+                                message_types=[self.message_type],
+                                auto_ack=False, offset='latest')
+
+    def _get_message(self):
+        """
+        Receive message from message bus.
+
+        Returns:
+            str: JSON object of message
+        """
+        try:
+            message = self.consumer.receive(timeout=0)
+            return json.loads(message.decode('utf-8'))
+        except Exception as e:
+            Log.error(f"Invalid format of message failed due to {e}. Message : {str(message)}")
```

Comment:
The error could be both: 1. failed to consume the message. 2. decode/load fails. I think lets update this error message to some generic string e.g. "Failed to get message, error{e}"

Reply:
Updated handling both saperatly